test/unit_tests/braket/experimental/autoqasm/test_pulse.py

```diff
@@ -183,3 +184,43 @@ def test_pulse_control_invalid_physical_qubit(instruction, qubits_or_frames, par
     with pytest.raises(ValueError):
         with aq.build_program():
             instruction(qubits_or_frames, *params)
+
+
+def test_pulse_freeparameter() -> None:
+    """Test pulse program with freeparameter."""
+
+    @aq.main
+    def my_program():
+        delay(["$3", "$4"], FreeParameter("duration"))
+
+    expected = textwrap.dedent(
+        """
+        OPENQASM 3.0;
+        input float[64] duration;
```

Comment:
This doesn't have to be a duration type?

Reply:
It gets converted to duration type via the `(duration) * 1s` expression below.

Reply:
Sweet, thanks! that makes sense